src/refs.c

```diff
@@ -214,52 +214,17 @@ int git_reference_lookup_resolved(
 	const char *name,
 	int max_nesting)
 {
-	git_refname_t scan_name;
-	git_reference_t scan_type;
-	int error = 0, nesting;
-	git_reference *ref = NULL;
+	git_refname_t normalized;
 	git_refdb *refdb;
+	int error = 0;
 
 	assert(ref_out && repo && name);
 
-	*ref_out = NULL;
-
-	if (max_nesting > MAX_NESTING_LEVEL)
-		max_nesting = MAX_NESTING_LEVEL;
-	else if (max_nesting < 0)
-		max_nesting = DEFAULT_NESTING_LEVEL;
```

Comment:
It looks like `DEFAULT_NESTING_LEVEL` is now unused and can get cleaned up.

`MAX_NESTING_LEVEL`, on the other hand, remains used in `get_terminal`.  🤔  Seems disappointing to have it defined twice.  Can we make it (or maybe both?) a global but internal-to-the-library constant?

Reply:
Oops, I should start doing proper reviews where the comments are batched until the end, because this question was obviously answered for me in the next commit.  🙃